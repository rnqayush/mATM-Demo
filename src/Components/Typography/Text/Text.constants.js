export const DEFAULT_FONT_SIZES = {
  default: 16,
  h1: 36,
  h2: 24,
  h3: 16,
};

export const HEADING_TAGS = Object.keys(DEFAULT_FONT_SIZES).filter((key) => key !== 'default');
