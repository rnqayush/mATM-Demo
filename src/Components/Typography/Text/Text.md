| Old(Deprecated) | New                                                       |
| --------------- | --------------------------------------------------------- |
| LargeHeading    | `<Text as="h1">Sample Text</Text>`                        |
| Heading         | `<Text as="h2">Sample Text</Text>`                        |
| Subheading      | `<Text as="h3">Sample Text</Text>`                        |
| Big             | `<Text fontSize={64}>Sample Text</Text>`                  |
| Paragraph       | `<Text lineHeight={20}>Sample Text</Text>`                |
| Title           | `<Text fontSize={20}><Strong>Sample Text</Strong></Text>` |
| Small           | `<Text fontSize={12} lineHeight={20}>Sample Text</Text>`  |
| Supersmall      | `<Text fontSize={10}>Sample Text</Text>`                  |

Text Without props

```jsx
<Text>Lorem Ipsum is simply dummy text of the printing and typesetting industry.</Text>
```

Text With `color` props

```jsx
<Text color="red">Lorem Ipsum is simply dummy text of the printing and typesetting industry.</Text>
```

Text With `Strong` Component

```jsx
import Strong from '../Strong';
<Text>
  Lorem Ipsum is <Strong>simply dummy</Strong> text of the printing and typesetting industry.
</Text>;
```

Text with different `as` (Heading) props

```jsx
<Text as="h1" >
  Lorem Ipsum is simply dummy text.
</Text>
<Text as="h2" >
  Lorem Ipsum is simply dummy text.
</Text>
<Text as="h3" >
  Lorem Ipsum is simply dummy text.
</Text>
```

Text with different `as` (Heading) and fontSize props

```jsx
<Text as="h1" fontSize={20}>
  Lorem Ipsum is simply dummy text.
</Text>
```

Text with different `align` prop.

```jsx
<Text align="left">Sample Text - left aligned (default)</Text>
<br />
<Text align="center">Sample Text - center aligned</Text>
<br />
<Text align="right">Sample Text - right aligned</Text>
```

Text with different `fontSize` prop.

```jsx
<Text fontSize={64}>Sample Text - 64px</Text>
<Text fontSize={36}>Sample Text - 36px</Text>
<Text fontSize={26}>Sample Text - 26px</Text>
<Text fontSize={24}>Sample Text - 24px</Text>
<Text fontSize={20}>Sample Text - 20px</Text>
<Text fontSize={16}>Sample Text - 16px</Text>
<Text fontSize={14}>Sample Text - 14px</Text>
<Text fontSize={12}>Sample Text - 12px</Text>
<Text fontSize={10}>Sample Text - 10px</Text>
```

Text With `padding` prop

```jsx
<Text color="red" paddingLeft={20}>This will have paddingLeft of 20</Text>
<Text color="red" paddingRight={20}>This will have paddingRight of 20</Text>
<Text color="red" paddingHorizontal={20}>This will have paddingHorizontal of 20</Text>
```

Text With `margin` prop

```jsx
<Text color="red" marginTop={10}>This will have marginTop of 10</Text>
<Text color="red" marginBottom={10}>This will have marginBottom of 10</Text>
<Text color="red" marginVertical={10}>This will have marginVertical of 10</Text>
```

Text with custom style and `as` prop.

```jsx
import styled from 'styled-components';

const FancyText = styled(Text)`
  color: #f00;
`;
<FancyText as="h1">Sample Text</FancyText>;
```
