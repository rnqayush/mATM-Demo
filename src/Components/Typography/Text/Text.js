import React from 'react';
import PropTypes from 'prop-types';
import * as COLOR from '@idfc/ccl-commons/palette/colors';

import {FontSize, LineHeight, TextAlign} from '@idfc/ccl-commons/enums';
import {TextWrapper} from './Text.style';
import {HEADING_TAGS} from './Text.constants';

const Text = ({
  as,
  color,
  fontSize,
  children,
  align,
  lineHeight,
  marginTop,
  marginRight,
  marginBottom,
  marginLeft,
  marginVertical,
  marginHorizontal,
  paddingTop,
  paddingRight,
  paddingBottom,
  paddingLeft,
  paddingVertical,
  paddingHorizontal,
  ...restProps
}) => (
  <TextWrapper
    color={color}
    fontSize={fontSize}
    align={align}
    type={as}
    marginTop={marginTop}
    marginRight={marginRight}
    marginBottom={marginBottom}
    marginLeft={marginLeft}
    marginVertical={marginVertical}
    marginHorizontal={marginHorizontal}
    paddingTop={paddingTop}
    paddingRight={paddingRight}
    paddingBottom={paddingBottom}
    paddingLeft={paddingLeft}
    paddingVertical={paddingVertical}
    paddingHorizontal={paddingHorizontal}
    lineHeight={lineHeight}
    {...restProps}>
    {children}
  </TextWrapper>
);

Text.propTypes = {
  /** Inner element is React component  */
  children: PropTypes.node,
  /** Text rendered with Default sizes  - (h1, h2, h3) */
  as: PropTypes.oneOf(HEADING_TAGS),
  /** Text rendered with given colors(from palette) - (maroon, red, green) etc... */
  color: PropTypes.oneOf(Object.keys(COLOR)),
  /** Text rendered with given font size - (10, 12, 14, 16, 20, 24, 26, 34, 64) */
  fontSize: PropTypes.oneOf(Object.values(FontSize)),
  /** Set the text-align attribute - {left, center, right} */
  align: PropTypes.oneOf(Object.values(TextAlign)),
  /** Set the line-height attribute - (18, 20, 22, 24) */
  lineHeight: PropTypes.oneOf(Object.values(LineHeight)),
  /** Set the margin attribute */
  marginTop: PropTypes.number,
  marginRight: PropTypes.number,
  marginBottom: PropTypes.number,
  marginLeft: PropTypes.number,
  marginVertical: PropTypes.number,
  marginHorizontal: PropTypes.number,
  /** Set the padding attribute */
  paddingTop: PropTypes.number,
  paddingRight: PropTypes.number,
  paddingBottom: PropTypes.number,
  paddingLeft: PropTypes.number,
  paddingVertical: PropTypes.number,
  paddingHorizontal: PropTypes.number,
  /** Sets font-size bold */
  bold: PropTypes.bool,
};

Text.defaultProps = {
  children: null,
  color: undefined,
  marginTop: undefined,
  marginRight: undefined,
  marginBottom: undefined,
  marginLeft: undefined,
  marginVertical: undefined,
  marginHorizontal: undefined,
  paddingTop: undefined,
  paddingRight: undefined,
  paddingBottom: undefined,
  paddingLeft: undefined,
  paddingVertical: undefined,
  paddingHorizontal: undefined,
  as: undefined,
  fontSize: undefined,
  align: undefined,
  lineHeight: undefined,
  bold: false,
};

/** @component */
export default Text;
