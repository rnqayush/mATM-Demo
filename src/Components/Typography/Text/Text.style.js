import styled from 'styled-components/native';
import COLORS from '@idfc/ccl-commons/palette';

import {DEFAULT_FONT_SIZES, HEADING_TAGS} from './Text.constants';

export const TextWrapper = styled.Text`
  color: ${({color, theme}) => (color && COLORS[color] ? theme?.Text?.FontColor[color] : theme?.Text?.FontColor?.grey500)};
  font-size: ${({fontSize, type}) => `${fontSize || DEFAULT_FONT_SIZES[type] || DEFAULT_FONT_SIZES.default}px`};
  font-family: ${({type, bold, theme}) => (bold || HEADING_TAGS.includes(type) ? theme?.FontFamily?.BOLD : theme?.FontFamily?.REGULAR)};

  ${({align}) => align && `text-align: ${align}`};
  ${({lineHeight}) => lineHeight && `line-height:${lineHeight}px`};

  ${({marginTop}) => marginTop && `margin-top:${marginTop}px`};
  ${({marginRight}) => marginRight && `margin-right:${marginRight}px`};
  ${({marginBottom}) => marginBottom && `margin-bottom:${marginBottom}px`};
  ${({marginLeft}) => marginLeft && `margin-left:${marginLeft}px`};
  ${({marginVertical}) => marginVertical && `marginVertical:${marginVertical}px`};
  ${({marginHorizontal}) => marginHorizontal && `marginHorizontal:${marginHorizontal}px`};

  ${({paddingTop}) => paddingTop && `padding-top:${paddingTop}px`};
  ${({paddingRight}) => paddingRight && `padding-right:${paddingRight}px`};
  ${({paddingBottom}) => paddingBottom && `padding-bottom:${paddingBottom}px`};
  ${({paddingLeft}) => paddingLeft && `padding-left:${paddingLeft}px`};
  ${({paddingVertical}) => paddingVertical && `paddingVertical:${paddingVertical}px`};
  ${({paddingHorizontal}) => paddingHorizontal && `paddingHorizontal:${paddingHorizontal}px`};
`;

export default TextWrapper;
