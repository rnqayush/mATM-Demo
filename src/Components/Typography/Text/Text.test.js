import React from 'react';
import {ThemeProvider} from 'styled-components/native';
import {FontColor} from '@idfc/ccl-commons/enums';

import {dark} from '../../../themes';
import {render, withConsoleErrorSpy} from '../../../../test/test-utils';

import Text from './index';

describe('text Component', () => {
  it('matches snapshot', () => {
    expect(render(<Text />)).toMatchSnapshot();
  });

  it('matches snapshot for dark mode', () => {
    expect(
      render(
        <ThemeProvider theme={dark}>
          <Text color={FontColor.GREY_400} />
        </ThemeProvider>,
      ),
    ).toMatchSnapshot();
  });

  it('when wrong color value passed in props, use default color', () => {
    withConsoleErrorSpy(1)(() => {
      const tree = render(<Text color="abcd" />).toJSON();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveStyleRule('color', '#54565b');
    });
  });

  it('matches snapshot with fontSize prop', () => {
    expect(render(<Text fontSize={12} />)).toMatchSnapshot();
  });

  it('matches snapshot with `bold` prop', () => {
    expect(render(<Text bold />)).toMatchSnapshot();
  });

  it('matches snapshot with as prop', () => {
    expect(render(<Text as="h1" />)).toMatchSnapshot();
  });

  it('matches snapshot with as prop and fontSize and lineHeight', () => {
    expect(render(<Text as="h1" fontSize={20} lineHeight={20} />)).toMatchSnapshot();
  });

  it('matches snapshot with as prop and margin and padding', () => {
    expect(render(<Text as="h1" marginTop={20} paddingLeft={10} lineHeight={24} />)).toMatchSnapshot();
  });

  it('matches snapshot with as prop and paddingVertical and marginHorizontal', () => {
    expect(render(<Text as="h1" marginHorizontal={20} paddingVertical={10} lineHeight={24} />)).toMatchSnapshot();
  });

  it('matches snapshot with all props', () => {
    expect(render(<Text as="h1" color="green" fontSize={24} align="right" lineHeight={24} />)).toMatchSnapshot();
  });
});
