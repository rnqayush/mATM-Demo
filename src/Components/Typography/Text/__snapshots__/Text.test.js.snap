// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text Component matches snapshot 1`] = `
<Text
  bold={false}
  style={
    Array [
      Object {
        "color": "#54565b",
        "fontFamily": "Roboto",
        "fontSize": 16,
      },
    ]
  }
/>
`;

exports[`text Component matches snapshot for dark mode 1`] = `
<Text
  bold={false}
  color="grey400"
  style={
    Array [
      Object {
        "color": "#7d7d7d",
        "fontFamily": "Roboto",
        "fontSize": 16,
      },
    ]
  }
/>
`;

exports[`text Component matches snapshot with \`bold\` prop 1`] = `
<Text
  bold={true}
  style={
    Array [
      Object {
        "color": "#54565b",
        "fontFamily": "Roboto-Bold",
        "fontSize": 16,
      },
    ]
  }
/>
`;

exports[`text Component matches snapshot with all props 1`] = `
<Text
  align="right"
  bold={false}
  color="green"
  fontSize={24}
  lineHeight={24}
  style={
    Array [
      Object {
        "color": "#00a07a",
        "fontFamily": "Roboto-Bold",
        "fontSize": 24,
        "lineHeight": 24,
        "textAlign": "right",
      },
    ]
  }
  type="h1"
/>
`;

exports[`text Component matches snapshot with as prop 1`] = `
<Text
  bold={false}
  style={
    Array [
      Object {
        "color": "#54565b",
        "fontFamily": "Roboto-Bold",
        "fontSize": 36,
      },
    ]
  }
  type="h1"
/>
`;

exports[`text Component matches snapshot with as prop and fontSize and lineHeight 1`] = `
<Text
  bold={false}
  fontSize={20}
  lineHeight={20}
  style={
    Array [
      Object {
        "color": "#54565b",
        "fontFamily": "Roboto-Bold",
        "fontSize": 20,
        "lineHeight": 20,
      },
    ]
  }
  type="h1"
/>
`;

exports[`text Component matches snapshot with as prop and margin and padding 1`] = `
<Text
  bold={false}
  lineHeight={24}
  marginTop={20}
  paddingLeft={10}
  style={
    Array [
      Object {
        "color": "#54565b",
        "fontFamily": "Roboto-Bold",
        "fontSize": 36,
        "lineHeight": 24,
        "marginTop": 20,
        "paddingLeft": 10,
      },
    ]
  }
  type="h1"
/>
`;

exports[`text Component matches snapshot with as prop and paddingVertical and marginHorizontal 1`] = `
<Text
  bold={false}
  lineHeight={24}
  marginHorizontal={20}
  paddingVertical={10}
  style={
    Array [
      Object {
        "color": "#54565b",
        "fontFamily": "Roboto-Bold",
        "fontSize": 36,
        "lineHeight": 24,
        "marginHorizontal": 20,
        "paddingVertical": 10,
      },
    ]
  }
  type="h1"
/>
`;

exports[`text Component matches snapshot with fontSize prop 1`] = `
<Text
  bold={false}
  fontSize={12}
  style={
    Array [
      Object {
        "color": "#54565b",
        "fontFamily": "Roboto",
        "fontSize": 12,
      },
    ]
  }
/>
`;

exports[`text Component when wrong color value passed in props, use default color 1`] = `
<Text
  bold={false}
  color="abcd"
  style={
    Array [
      Object {
        "color": "#54565b",
        "fontFamily": "Roboto",
        "fontSize": 16,
      },
    ]
  }
/>
`;
